src/Microsoft.VisualStudio.Threading.Shared/AsyncBarrier.cs

```diff
@@ -25,24 +25,22 @@ public class AsyncBarrier
         private readonly int participantCount;
 
         /// <summary>
-        /// The number of participants that have not yet signaled the barrier.
-        /// </summary>
-        private int remainingParticipants;
-
-        /// <summary>
         /// The set of participants who have reached the barrier, with their awaiters that can resume those participants.
         /// </summary>
-        private ConcurrentStack<TaskCompletionSource<bool>> waiters;
+        private Stack<TaskCompletionSource<EmptyStruct>> waiters;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="AsyncBarrier"/> class.
         /// </summary>
         /// <param name="participants">The number of participants.</param>
         public AsyncBarrier(int participants)
         {
-            Requires.Range(participants > 0, "participants");
-            this.remainingParticipants = this.participantCount = participants;
-            this.waiters = new ConcurrentStack<TaskCompletionSource<bool>>();
+            Requires.Range(participants > 0, nameof(participants));
```

Comment:
Good question. :) But as this has already shipped, we can't make it more restrictive now.